src/redux/features/user/usersAsyncThunks.ts

```diff
@@ -34,11 +34,15 @@ export const requestSearchUserByEmail = createAsyncThunk(
 
 export const requestUserCookieConsentUpdate = createAsyncThunk(
   'users/cookie-consent/update',
-  async (payload: UpdateUserCookieConsentRequestPayload, { rejectWithValue }) => {
+  async (
+    { payload, reloadUser }: { payload: UpdateUserCookieConsentRequestPayload; reloadUser: () => void },
+    { rejectWithValue }
+  ) => {
```

Comment:
This works as-is, but I'm happy to `reloadUser` elsewhere if we'd prefer that in another location.

Reply:
Yea it seems a bit squirrelly to have the `CookieConsentBanner` component be the glue between redux and a provider. 

It is a bit more work, but it seems like the `dispatch(requestUserCookieConsentUpdate(...` should be inside the `UserProvider`. This way it can look at the response and refresh itself when the response comes back.

Reply:
Yeah I agree it wasn't the cleanest. I reverted the changes in the thunk and opted to call `reloadUser` after awaiting the thunk dispatch, which is in-line with how we're reloading the user elsewhere in the app.

Reply:
Moving the function into the `UserProvider` now...

Reply:
Sorry, I didn't mean to hold this up! It can be followed up on later